src/sys/socket/addr.rs

```diff
@@ -15,12 +17,24 @@ use std::os::unix::ffi::OsStrExt;
  *
  */
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+#[repr(C)]
+pub struct sockaddr_nl {
```

Comment:
I also just realised this would mean none of those impls would be present. unclear if they're required or not?